std.os.setsockopt broken with libc

`std.os.setsockopt` appears to be broken when compiling with libc.

Example:

```zig
usingnamespace @import("std");

pub fn main() !void {
    const fd = try os.socket(os.AF_INET, os.SOCK_STREAM, 0);
    _ = try os.setsockopt(fd, os.SOL_SOCKET, os.SO_REUSEADDR, &mem.toBytes(@as(c_int, 1)));

    // The same bug breaks StreamServer:

    var server = net.StreamServer.init(.{});
    try server.listen(try net.Address.parseIp4("127.0.0.1", 12345));
}
```

This compiles fine with `zig build-exe main.zig` but gives the following error when compiling with `--library c`:

```
/nix/store/syj90d418f63szyinbklp69zvv10v8qy-zig-5acc8afb5f9674b9b7b290635e9c2837872b8a93/lib/zig/std/os.zig:4228:60: error: cast discards const qualifier
    switch (errno(system.setsockopt(fd, level, optname, opt.ptr, @intCast(socklen_t, opt.len)))) {
                                                           ^
```

I got around this issue by removing the const qualifier in my copy of the standard library:

```diff
- pub fn setsockopt(fd: fd_t, level: u32, optname: u32, opt: []const u8) SetSockOptError!void {
+ pub fn setsockopt(fd: fd_t, level: u32, optname: u32, opt: []u8) SetSockOptError!void {
```

I'm not sure if that's an ok fix though so I did not open a PR.

--- a/lib/std/c.zig
+++ b/lib/std/c.zig
@@ -135,12 +135,13 @@ pub extern "c" fn uname(buf: *utsname) c_int;
 pub extern "c" fn gethostname(name: [*]u8, len: usize) c_int;
 pub extern "c" fn bind(socket: fd_t, address: ?*const sockaddr, address_len: socklen_t) c_int;
 pub extern "c" fn socket(domain: c_uint, sock_type: c_uint, protocol: c_uint) c_int;
+pub extern "c" fn socketpair(domain: c_uint, sock_type: c_uint, protocol: c_uint, sv: *[2]fd_t) c_int;
 pub extern "c" fn listen(sockfd: fd_t, backlog: c_uint) c_int;
 pub extern "c" fn getsockname(sockfd: fd_t, noalias addr: *sockaddr, noalias addrlen: *socklen_t) c_int;
 pub extern "c" fn connect(sockfd: fd_t, sock_addr: *const sockaddr, addrlen: socklen_t) c_int;
 pub extern "c" fn accept4(sockfd: fd_t, addr: *sockaddr, addrlen: *socklen_t, flags: c_uint) c_int;
-pub extern "c" fn getsockopt(sockfd: fd_t, level: u32, optname: u32, optval: *c_void, optlen: *socklen_t) c_int;
-pub extern "c" fn setsockopt(sockfd: fd_t, level: u32, optname: u32, optval: *c_void, optlen: socklen_t) c_int;
+pub extern "c" fn getsockopt(sockfd: fd_t, level: u32, optname: u32, optval: ?*c_void, optlen: *socklen_t) c_int;
+pub extern "c" fn setsockopt(sockfd: fd_t, level: u32, optname: u32, optval: ?*const c_void, optlen: socklen_t) c_int;
 pub extern "c" fn send(sockfd: fd_t, buf: *const c_void, len: usize, flags: u32) isize;
 pub extern "c" fn sendto(
     sockfd: fd_t,

